stock_picking_actual_date/models/stock_picking.py

```diff
@@ -20,17 +20,36 @@ def _compute_is_editable_actual_date(self):
             ):
                 rec.is_editable_actual_date = True
 
+    def _sequence_actual_date_matches(self, move, date):
+        sequence = move.name
+        format_values = move._get_sequence_format_param(sequence)[1]
+        sequence_number_reset = move._deduce_sequence_number_reset(sequence)
+        year_start, year_end = move._get_sequence_date_range(sequence_number_reset)
+        year_match = (
+            not format_values["year"]
+            or move._year_match(format_values["year"], year_start)
+        ) and (
+            not format_values["year_end"]
+            or move._year_match(format_values["year_end"], year_end)
+        )
+        month_match = not format_values["month"] or format_values["month"] == date.month
+        return year_match and month_match
+
     def write(self, vals):
         res = super().write(vals)
-        if "actual_date" in vals:
+        if "actual_date" in vals and vals["actual_date"]:
             for rec in self:
                 if rec.state != "done":
                     continue
                 account_moves = rec.move_ids.account_move_ids
                 if not account_moves:
                     continue
                 account_moves.button_draft()
-                account_moves.name = False
-                account_moves.date = rec.actual_date
+                for move in account_moves:
+                    if self._sequence_actual_date_matches(move, rec.actual_date):
+                        move.date = rec.actual_date
+                    else:
+                        move.name = False
+                        move.date = rec.actual_date
```

Comment:
Could we do something like this, obviously extending `_constrains_date_sequence()` to handle the context as well?

```suggestion
                    move.with_context(skip_date_sequence_constraint=True)
                        move.date = rec.actual_date
                        if not move._sequence_matches_date():
                            move.name = False
```

Reply:
> extending _constrains_date_sequence() to handle the context

We can actually extend `_must_check_constrains_date_sequence()` instead.

Reply:
I extended `_must_check_constrains_date_sequence()` instead of `_constrains_date_sequence()` because I believe this method is for the purpose of skipping or not skipping check.

Reply:
> We can actually extend _must_check_constrains_date_sequence() instead.

Sorry. My browser is not refreshed and I didn't see your last comment but my idea was the same.